lookout/style/format/feature_extractor.py

```diff
@@ -348,21 +386,17 @@ def _add_noops(vnodes: Sequence[VirtualNode], path: str) -> List[VirtualNode]:
                                       y=CLASS_INDEX[CLS_NOOP], path=path))
         return result
 
-    @staticmethod
-    def _index_nodes(vnodes: Sequence[VirtualNode]) -> None:
-        global_index = 0
-        labeled_index = 0
-        for vnode in vnodes:
-            vnode.global_index = global_index
-            global_index += 1
-            if vnode.y is not None:
-                vnode.labeled_index = labeled_index
-                labeled_index += 1
-
-    def _get_features(self, features: Sequence[str], sibling: Union[VirtualNode, bblfsh.Node],
-                      node: VirtualNode) -> List[int]:
-        for feature in features:
-            yield from self._features[feature](sibling, node)
+    def _get_features(self, feature_group: FeatureGroup, repetition: int,
+                      sibling: Union[VirtualNode, bblfsh.Node], node: VirtualNode
+                      ) -> Iterable[int]:
+        for feature_name, indices in self.feature_to_index[feature_group][repetition].items():
+            if not len(indices):
+                continue
+            indices_set = set(indices)
+            yield from (result for i, result in enumerate(
+                    self._features[feature_name](sibling, node),
+                    start=self._feature_offsets[feature_group][repetition][feature_name])
+                if i in indices_set)
```

Comment:
The goal here is to avoid calling select_features() when you load a model (which I'm strongly against since it's super error-prone). This is the price to pay. I can for sure cache the sets though.

Reply:
ok